packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -344,12 +372,16 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
         }
 
         const changes = this.popContentChanges();
-        if (changes.length === 0) {
+        if (changes.length === 0 && overwriteEncoding === undefined) {
             return;
         }
 
         const content = this.model.getValue();
-        await Resource.save(this.resource, { changes, content }, token);
+        if (this.getEncoding() || overwriteEncoding) {
+            await Resource.save(this.resource, { changes, content, options: { encoding: this.getEncoding(), overwriteEncoding } }, token);
```

Comment:
Why not just this line? `this.getEncoding` is undefined then `encoding` will be undefined, the same about `overwriteEncoding`

Reply:
True! Fixed now.